libs/dataset_deployer.py

```diff
@@ -71,26 +71,23 @@ def flatten_dict(data: dict, level_separator: str = ".") -> dict:
 
 
 def write_nested_csv(
-    data: List[dict], output_path: pathlib.Path, keys_to_skip: Optional[List[str]] = None
+    data: List[dict], output_path: pathlib.Path, header: Optional[List[str]] = None,
 ):
     """Writes list of data as a nested csv.
 
     Args:
         data: list of data to write.
         output_path: Path of file to write to.
-        keys_to_skip: Keys to skip.  Keys can be flattened entries or top level keys.
-
+        header: Columns to output.  If specified, header will be in order of columns.
     """
-    keys_to_skip = keys_to_skip or []
-
     if not data:
         raise ValueError("Cannot upload a 0 length list.")
 
     first_row = data[0]
-    first_row = {key: value for key, value in first_row.items() if key not in keys_to_skip}
+    first_row = {key: value for key, value in first_row.items()}
 
-    header = flatten_dict(first_row).keys()
-    header = [column for column in header if column not in keys_to_skip]
+    all_columns = list(flatten_dict(first_row).keys())
+    header = header or all_columns
```

Comment:
not fooling me, you've just hidden the branch :-P More seriously, why maintain functionality that isn't used in production? when refactoring I want to quickly see all the ways a function is called by non-test code. It is nice to get that information from the function prototype without manually inspecting each call site. If keeping `header` as `Optional` is useful for tests or future development please document it.

Reply:
I definitely see your point - I think i prefer the optional argument here because this function is primarily meant to just write data in a list of dicts to a csv.  If i were looking at that function, I would expect to be able to just write the csv by passing the dictionary and a file path.  I think if we make the `header` (or could change it to `header_override`) required, I would change the name of the function to indicate that that's write nested csv in specific heading order.  

I dont' think that's really necessary, so I don't see it as much as additional functionality - more as api fluency.

Reply:
I'd rather see the awkward reality instead of a more elegant alternative.... though I have a hunch that isn't how you see the two choices here.